backend/src/test/java/heartbeat/service/report/ReportServiceTest.java

```diff
@@ -104,4 +101,25 @@ void ShouldInitializeMetricsDataCompletedInHandlerWithPreOneWhenPreOneExisted()
 		verify(generateReporterService, never()).generateDoraReport(request);
 	}
 
+	@Test
+	void ShouldInitializeMetricsDataCompletedInHandlerWhenRequestMetricsExist() throws InterruptedException {
+		GenerateReportRequest request = GenerateReportRequest.builder()
+			.csvTimeStamp("csvTimeStamp")
+			.metrics(List.of("velocity", "lead time for changes"))
+			.build();
+		when(asyncMetricsDataHandler.getMetricsDataCompleted(any()))
+			.thenReturn(MetricsDataCompleted.builder().pipelineMetricsCompleted(true).build());
+		doAnswer(invocation -> null).when(generateReporterService).generateBoardReport(request);
+		reportService.generateReportByType(request, MetricType.BOARD);
+		Thread.sleep(100);
```

Comment:
try to remove sleep to improve the speed of UT
Maybe you can mock more things

Reply:
done